### ISSUE TITLE
spurious error in thread pool executor submission w/ *args function

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2843]
**Describe the bug**
pyright has a spurious error in threadpoolexecutor's submit function when using a function with an `*args` parameter

**To Reproduce**
```python
from typing import Any
import concurrent.futures


def args_func(name: str, *args: Any):
    ...


with concurrent.futures.ThreadPoolExecutor() as executor:
    fut = executor.submit(args_func, "hello", "a", "b")  # expected 2 positional arguments
```

The error is incorrect.

**Expected behavior**
The above code snippet should type check/not given an error that it expects two positional arguments with the executor submission.

**VS Code extension or command-line**
Linux commandline pyright 1.1.206

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this is a bug, but I don't think it's a regression. I've confirmed that the same error occurs in older versions of pyright too.

**Comment by GBeauregard** [comment]
> Thanks for the bug report. I agree this is a bug, but I don't think it's a regression. I've confirmed that the same error occurs in older versions of pyright too.

I assumed it was a regression because the behavior was new, but upon investigation it seems the newness is because typeshed was made correct and not because of a regression in pyright: https://github.com/python/typeshed/commit/fe75ec69bfddb9bdb3c3d2e32f769a84c919d80a

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.207, which I just published. It will also be included in the next release of pylance.
